pkg/controllers/machinesetsync/machineset_sync_controller_test.go

```diff
@@ -156,56 +155,326 @@ var _ = Describe("With a running MachineSetSync controller", func() {
+				Expect(k8sClient.Create(ctx, mapiMachineSet)).Should(Succeed())
+
+				By("Setting the AuthoritativeAPI to Migrating")
+				Eventually(k.UpdateStatus(mapiMachineSet, func() {
+					mapiMachineSet.Status.AuthoritativeAPI = machinev1beta1.MachineAuthorityMigrating
+				})).Should(Succeed())
+			})
+
+			It("should not make any changes to either machineset", func() {
+				mapiResourceVersion := mapiMachineSet.GetResourceVersion()
+				capiResourceVersion := capiMachineSet.GetResourceVersion()
+				Consistently(k.Object(mapiMachineSet), timeout).Should(
+					HaveField("ResourceVersion", Equal(mapiResourceVersion)),
+				)
+				Consistently(k.Object(capiMachineSet), timeout).Should(
+					HaveField("ResourceVersion", Equal(capiResourceVersion)),
+				)
+			})
+		})
+
+		Context("When the MAPI MachineAuthority is not set", func() {
+			BeforeEach(func() {
+				By("Creating the CAPI and MAPI MachineSets")
+				mapiMachineSet = mapiMachineSetBuilder.Build()
+				capiMachineSet = capiMachineSetBuilder.Build()
+
+				Expect(k8sClient.Create(ctx, capiMachineSet)).Should(Succeed())
+				Expect(k8sClient.Create(ctx, mapiMachineSet)).Should(Succeed())
+
+				By("Setting the AuthoritativeAPI to Migrating")
+				Eventually(k.UpdateStatus(mapiMachineSet, func() {
+					mapiMachineSet.Status.AuthoritativeAPI = ""
+				})).Should(Succeed())
+			})
+
+			// I think there may be another case here, where we have had it previously set - synchronized and then it gets unset.
+			// We may want to ensure that we don't change resourceVersion, and that the condition does not change?
+			// Changes to false with a failure reason?
+			//
+			// I'm not sure if we have an API validation that stops this.
+			// TODO: clarify / work out
+
+			It("should not make any changes", func() {
+				resourceVersion := mapiMachineSet.GetResourceVersion()
+				Consistently(k.Object(mapiMachineSet), timeout).Should(
+					HaveField("ResourceVersion", Equal(resourceVersion)),
+				)
+			})
+		})
+
+		Context("when the MAPI MachineSet does not exist and the CAPI MachineSet does", func() {
+			BeforeEach(func() {
+				By("Creating the CAPI MachineSet")
+				capiMachineSet = capiMachineSetBuilder.Build()
+
+				Expect(k8sClient.Create(ctx, capiMachineSet)).Should(Succeed())
+			})
+
+			It("should not make any changes to the CAPI MachineSet", func() {
+				resourceVersion := capiMachineSet.GetResourceVersion()
+				Consistently(k.Object(capiMachineSet), timeout).Should(
+					HaveField("ResourceVersion", Equal(resourceVersion)),
+				)
+			})
+
+			// I can't work out a nice way to write this as I'm tired :)
+
+			// It("should not create a MAPI MachineSet", func() {
+			// 	Consistently(k.ObjectList(&machinev1beta1.MachineSetList{}), timeout).ShouldNot(
+			// 		ContainElement(WithTransform(
+			// 			func(ms machinev1beta1.MachineSet) string {
+			// 				return ms.GetName()
+			// 			}, Equal(capiMachineSet.GetName()))))
+			// })
```

Comment:
I think something along the lines of
```suggestion
			 It("should not create a MAPI MachineSet", func() {
			 	Consistently(k.ObjectList(&machinev1beta1.MachineSetList{}), timeout).ShouldNot(
			 			ContainElement(HaveField("ObjectMeta.Name", Equal(capiMachineSet.GetName()))),
			 		),
			 	)
			 })
```

Reply:
Ended up with:
```go
It("should not create a MAPI MachineSet", func() {
  Consistently(k.ObjectList(&machinev1beta1.MachineSetList{}), timeout).ShouldNot(HaveField("Items",
	  ContainElement(HaveField("ObjectMeta.Name", Equal(capiMachineSet.GetName()))),
  ))
})
```